packages/eslint-plugin/src/rules/no-mixed-enums.ts

```diff
@@ -0,0 +1,222 @@
+      return tsutils.isTypeFlagSet(
+        typeChecker.getTypeAtLocation(node),
+        ts.TypeFlags.StringLike,
+      )
+        ? ts.TypeFlags.String
+        : ts.TypeFlags.Number;
+    }
+
+    function getTypeFromImported(
+      imported: TSESTree.Node,
+    ): AllowedType | undefined {
+      const type = typeChecker.getTypeAtLocation(
+        parserServices.esTreeNodeToTSNodeMap.get(imported),
+      );
+
+      const valueDeclaration = type.getSymbol()?.valueDeclaration;
+      if (
+        !valueDeclaration ||
+        !ts.isEnumDeclaration(valueDeclaration) ||
+        valueDeclaration.members.length === 0
+      ) {
+        return undefined;
+      }
+
+      return getAllowedTypeForNode(valueDeclaration.members[0]);
+    }
+
+    function getMemberType(
+      member: TSESTree.TSEnumMember,
+    ): AllowedType | ts.TypeFlags.Unknown | undefined {
+      if (!member.initializer) {
+        return undefined;
+      }
+
+      switch (member.initializer.type) {
+        case AST_NODE_TYPES.Literal:
+          switch (typeof member.initializer.value) {
+            case 'number':
+              return ts.TypeFlags.Number;
+            case 'string':
+              return ts.TypeFlags.String;
+            default:
+              return ts.TypeFlags.Unknown;
+          }
+
+        case AST_NODE_TYPES.TemplateLiteral:
+          return ts.TypeFlags.String;
+
+        default:
+          return getAllowedTypeForNode(
+            parserServices.esTreeNodeToTSNodeMap.get(member.initializer),
+          );
+      }
+    }
+
+    function getDesiredTypeForDefinition(
+      node: TSESTree.TSEnumDeclaration,
+    ): AllowedType | ts.TypeFlags.Unknown | undefined {
+      const { imports, previousSibling } = collectNodeDefinitions(node);
+
+      // Case: Merged ambiently via module augmentation
+      // import { MyEnum } from 'other-module';
+      // declare module 'other-module' {
+      //   enum MyEnum { A }
+      // }
+      for (const imported of imports) {
+        const typeFromImported = getTypeFromImported(imported);
+        if (typeFromImported !== undefined) {
+          return typeFromImported;
+        }
+      }
+
+      // Case: Multiple enum declarations in the same file
+      // enum MyEnum { A }
+      // enum MyEnum { B }
+      if (previousSibling) {
+        return getMemberType(previousSibling.members[0]);
+      }
+
+      // Case: Namespace declaration merging
+      // namespace MyNamespace {
+      //   export enum MyEnum { A }
+      // }
+      // namespace MyNamespace {
+      //   export enum MyEnum { B }
+      // }
+      if (
+        node.parent!.type === AST_NODE_TYPES.ExportNamedDeclaration &&
+        node.parent!.parent!.type === AST_NODE_TYPES.TSModuleBlock
+      ) {
+        // TODO: We don't need to dip into the TypeScript type checker here!
+        // Merged namespaces must all exist in the same file.
+        // We could instead compare this file's nodes to find the merges.
```

Comment:
@bradzacher I poked around locally at writing this and got it to quite a few dozen lines of code to figure out nested namespaces in the same file [[example nested namespaces playground](https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&showAST=es&code=HYQwtgpgzgDiDGEAEB5ArgFwgJyQbwCgkkIAPGAe2wyVElgWQElhgd8jiTyqaJg0YJABVoNPEgCCSALxIA5JPlIAvpzVqCdaHESpM7QsTKVqtcDsZIWbXEa4neJAUNFRxSAEKyFn5WuINAgJ0LGwAOhsccLcMcMkAbhCDCKiI2PDPBKA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA)]. I suspect this is a pretty rare case and probably fine to treat as a followup performance improvement issue. WDYT?

Reply:
Actually - ignore me. This won't work because you can infinitely nest namespaces like:
```ts
namespace Foo {
  namespace Bar {
    namespace Baz {
      export enum FFS { }
    }
  }
}
namespace Foo {
  namespace Bar {
    namespace Baz {
      export enum FFS { }
    }
  }
}
```

So  I think this simple check for just "if child of a namespace, assume might merge" is good enough.